common/cuda_hip/solver/lobpcg_kernels.cpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/lobpcg_kernels.hpp"
+
+#include <ginkgo/core/base/types.hpp>
+
+#include "common/cuda_hip/base/dev_lapack_bindings.hpp"
+
+#if GKO_HAVE_LAPACK
+
+
+namespace gko {
+namespace kernels {
+namespace GKO_DEVICE_NAMESPACE {
+namespace lobpcg {
+
+
+using gko::kernels::lobpcg::workspace_mode;
+
+template <typename ValueType>
+void symm_generalized_eig(std::shared_ptr<const DefaultExecutor> exec,
+                          const workspace_mode alloc,
+                          matrix::Dense<ValueType>* a,
+                          matrix::Dense<ValueType>* b,
+                          array<remove_complex<ValueType>>* e_vals,
+                          array<char>* workspace)
+{
+    const auto id = exec->get_device_id();
+    auto handle = exec->get_dev_lapack_handle();
+
+    int n = static_cast<int>(a->get_size()[1]);  // column-major
+    int lda = static_cast<int>(a->get_stride());
+    int ldb = static_cast<int>(b->get_stride());
+    int fp_buffer_num_elems;
+    if (alloc == workspace_mode::allocate) {
+        dev_lapack::sygvd_buffersize(handle, LAPACK_EIG_TYPE_1,
+                                     LAPACK_EIG_VECTOR, LAPACK_FILL_LOWER, n,
+                                     a->get_values(), lda, b->get_values(), ldb,
+                                     e_vals->get_data(), &fp_buffer_num_elems);
+        size_type total_bytes = sizeof(ValueType) * fp_buffer_num_elems;
+        workspace->resize_and_reset(total_bytes);
+    } else {
+        fp_buffer_num_elems = workspace->get_size() / sizeof(ValueType);
+    }
+    array<int> dev_info(exec, 1);
```

Comment:
handle the dev_info result in the end.

Reply:
I added something in 41b80cc4b716074d4ab1f963445838fa970bebb3, is this like what you were thinking?

Reply:
originally I am thinking like copy the value to host and do the same thing like lapack by checking the value.
yes, yours should be good. not sure how the others think about the another catch in the call.

Reply:
you also need to throw the error again.

Reply:
Do I need to throw the error again, or just add a `std::exit` to the catch block? I'm already printing the message of the exception (because I wanted that to print before `devInfo`, originally). I could also use `devInfo` as the error code for the exit? Or would it be better to go with something generic like `-1`?

Reply:
We do not print the error message / exit from the library.

Reply:
I tried another version in 87e0d92048fc89b87df1caec0d1bb139171a8f04. I'm still not thrilled with it, but I'm not sure how else to print the `devInfo` value from cuSOLVER both when it gives an error (e.g., reporting which parameter is invalid) and when it doesn't (failure of routine).